src/Raven.Server/Commercial/LicenseHandler.cs

```diff
@@ -0,0 +1,183 @@
+        
+        private static RSAParameters RSAParameters
+        {
+            get
+            {
+                if (_rsaParameters != null)
+                    return _rsaParameters.Value;
+
+                string publicKeyString;
+                const string publicKeyPath = "Raven.Server.Commercial.RavenDB.public.json";
+                using (var stream = typeof(LicenseHandler).GetTypeInfo().Assembly.GetManifestResourceStream(publicKeyPath))
+                {
+                    if (stream == null)
+                        throw new InvalidOperationException("Could not find public key for the license");
+                    publicKeyString = new StreamReader(stream).ReadToEnd();
+                }
+
+                var rsaPublicParameters = JsonConvert.DeserializeObject<RSAPublicParameters>(publicKeyString);
+                _rsaParameters = new RSAParameters
+                {
+                    Modulus = rsaPublicParameters.RsaKeyValue.Modulus,
+                    Exponent = rsaPublicParameters.RsaKeyValue.Exponent
+                };
+                return _rsaParameters.Value;
+            }
+        }
+
+        public LicenseHandler(ServerStore serverStore)
+        {
+            _serverStore = serverStore;
+            _httpClient = new HttpClient
+            {
+                BaseAddress = new Uri(ApiRavenDbNet)
+            };
+
+            /*_leaseLicenseTimer = new Timer((state) =>
+                AsyncHelpers.RunSync(LeaseLicense), null, 0, (int)TimeSpan.FromHours(24).TotalMilliseconds);*/
+        }
+
+        public void Initialize()
+        {
+            var firstServerStartDate = _serverStore.LicenseStorage.GetFirstServerStartDate();
+            if (firstServerStartDate == null)
+            {
+                firstServerStartDate = DateTime.UtcNow;
+                _serverStore.LicenseStorage.SetFirstServerStartDate(firstServerStartDate.Value);
+            }
+
+            FirstServerStartDate = firstServerStartDate.Value;
+
+            var license = _serverStore.LicenseStorage.LoadLicense();
+            if (license == null)
+                return;
+
+            try
+            {
+                LicenseStatus.Attributes = LicenseValidator.Validate(license, RSAParameters);
+                LicenseStatus.Error = false;
+                LicenseStatus.Message = null;
+            }
+            catch (Exception e)
+            {
+                LicenseStatus.Attributes = null;
+                LicenseStatus.Error = true;
+                LicenseStatus.Message = e.Message;
+
+                if (Logger.IsInfoEnabled)
+                    Logger.Info("Could not validate license", e);
+
+                throw new InvalidDataException("Could not validate license!");
+            }
+        }
+
+        public static LicenseStatus GetLicenseStatus()
+        {
+            return LicenseStatus;
+        }
+
+        public static async Task Register(RegisteredUserInfo registeredUserInfo)
+        {
```

Comment:
This doesn't actually do registration. It just validate this against remote server.
We need to be able to update the license itself. Customer may have purchased additional things, or dates have changed, etc.